portfolio/src/main/webapp/script.js

```diff
@@ -105,3 +134,47 @@ function createCommentNode(comment) {
 
   return comDiv;
 }
+
+function onSignIn(googleUser) {
+  const profile = googleUser.getBasicProfile();
+  const id_token = googleUser.getAuthResponse().id_token;
+
+  let request = new XMLHttpRequest();
+  request.open("POST", "/auth");
+  request.setRequestHeader("Content-Type", "application/x-www-form-urlencoded");
+  request.onload = function() {
+    const text = request.responseText;
+
+    if (text !== null) {
+      user = JSON.parse(text);
+      user.id_token = id_token;
+      if (user.email !== profile.getEmail()) {
+        signOut();
+      }
+      toggleCommentForm();
+    } 
+  };
+  request.send("id_token=" + id_token);
+}
+
+function toggleCommentForm() {
```

Comment:
Can you also add information about which account is the visitor currently signed into?

If you do this in this method, you can rename it to e.g. `updateSignedInState`

Reply:
I don't think I understand the question - doesn't the google sign-in button automatically send the current email & account of the visitor to the script?

Reply:
Yes, but the script knows, but the page doesn't communicate to the visitor which account is signed in. (It's just an improvement suggestion, I think that you can freely decide not to do this in this project.)